### PR TITLE
Add bastion table

### PR DESCRIPTION
**Pull request by scotti-fletcher**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
select 
*
from oci_bastion_session 
where bastion_id = 'ocid1.bastion.oc1.ap-sydney-1.'
and target_resource_details->'sessionType' = '"MANAGED_SSH"'


select * from oci_bastion_session where 
id = 'ocid1.bastionsession.oc1.ap-sydney-1.'

select * from oci_bastion_bastion
where id = 'ocid1.bastion.oc1.ap-sydney-1.'

select 
*
FROM oci_bastion_bastion where (client_cidr_block_allow_list)::jsonb ? '0.0.0.0/0'

SELECT * from oci_bastion_bastion where id = 'ocid1.bastion.oc1.ap-sydney-1.'

select s.* from oci_bastion_bastion b
INNER JOIN oci_bastion_session s
ON b.id = s.bastion_id

</details>

### CODE REVIEW

**Comment by misraved** [comment]
@scotti-fletcher could you please fix the merge conflicts across all the open PRs? Could you also format the queries in the table docs using https://www.freeformatter.com/sql-formatter.html

File: oci/table_oci_bastion_session.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	// handle empty keyId, endpoint and region in list call
+	if bastionId == "" {
+		return nil, nil
+	}
+
+	// Create Session
+	session, err := bastionService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
```
Comment (misraved) [inline oci/table_oci_bastion_session.go:151]:
```suggestion
	if err != nil {
	       plugin.Logger(ctx).Error("oci_bastion_session.listBastionSessions", "connection_error", err")
		return nil, err
	}
```

File: oci/table_oci_bastion_session.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			request.Limit = types.Int(int(*limit))
+		}
+	}
+
+	pagesLeft := true
+	for pagesLeft {
+		response, err := session.BastionClient.ListSessions(ctx, request)
+		if err != nil {
+			return nil, err
+		}
```
Comment (misraved) [inline oci/table_oci_bastion_session.go:178]:
```suggestion
		if err != nil {
		        plugin.Logger(ctx).Error("oci_bastion_session.listBastionSessions", "api_error", err")
			return nil, err
		}
```

File: oci/table_oci_bastion_session.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+	// handle empty id in get call
+	if id == "" {
+		return nil, nil
+	}
+
+	// Create Session
+	session, err := bastionService(ctx, d, region)
+	if err != nil {
+		logger.Error("getBastionSession", "error_BastionService", err)
```
Comment (misraved) [inline oci/table_oci_bastion_session.go:220]:
```suggestion
		logger.Error("oci_bastion_session.getBastionSession", "connection_error", err)
```

File: oci/table_oci_bastion_session.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+		RequestMetadata: common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(d.Connection),
+		},
+	}
+
+	response, err := session.BastionClient.GetSession(ctx, request)
+
+	if err != nil {
+		return nil, err
+	}
```
Comment (misraved) [inline oci/table_oci_bastion_session.go:235]:
```suggestion
	if err != nil {
	       logger.Error("oci_bastion_session.getBastionSession", "api_error", err)
		return nil, err
	}
```

File: oci/table_oci_bastion_session.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+	return nil, err
+}
+
+// // HYDRATE FUNCTION
+func getBastionSession(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	logger.Debug("getBastionSession", "OCI_REGION", region)
```
Comment (misraved) [inline oci/table_oci_bastion_session.go:202]:
```suggestion

```

**Review by misraved — COMMENTED** [review]
@scotti-fletcher thanks for this new table and apologies for the delayed review. I observed that other tables will require similar changes.

Summary of changes:
1. The SQL queries in the table docs need to be formatted using https://www.freeformatter.com/sql-formatter.html
2. We need to add the missing logging information at the end of every API call. The format is roughly like - `plugin.Logger(ctx).Error("{table_name}.{function_name}", "connection_error/api_error", err)`

Thank you once again for the great work on the tables 👍.

File: oci/table_oci_bastion_session.go (hunk 5 above)
Comment (scotti-fletcher) [reply oci/table_oci_bastion_session.go:202]: Hi @misraved, Sure I can make those changes. This PR was done before I created the table generator.

Also just FYI There's another PR that contains all the changes to the SQL formatting. It's #454

**Pull request merged**